wallet/core/src/storage/local/mod.rs

```diff
@@ -10,8 +22,65 @@ pub mod wallet;
 pub use collection::Collection;
 pub use payload::Payload;
 pub use storage::Storage;
-pub use wallet::Wallet;
+pub use wallet::WalletStorage;
+
+use crate::error::Error;
+use crate::result::Result;
+use wasm_bindgen::prelude::*;
+use workflow_store::fs::create_dir_all_sync;
+
+static mut DEFAULT_STORAGE_FOLDER: Option<String> = None;
+static mut DEFAULT_WALLET_FILE: Option<String> = None;
+static mut DEFAULT_SETTINGS_FILE: Option<String> = None;
+
+pub fn default_storage_folder() -> &'static str {
+    unsafe { DEFAULT_STORAGE_FOLDER.get_or_insert("~/.kaspa".to_string()).as_str() }
+}
+
+pub fn default_wallet_file() -> &'static str {
+    unsafe { DEFAULT_WALLET_FILE.get_or_insert("kaspa".to_string()).as_str() }
+}
+
+pub fn default_settings_file() -> &'static str {
+    unsafe { DEFAULT_SETTINGS_FILE.get_or_insert("kaspa".to_string()).as_str() }
+}
+
+/// Set a custom storage folder for the wallet SDK
+/// subsystem.  Encrypted wallet files and transaction
+/// data will be stored in this folder. If not set
+/// the storage folder will default to `~/.kaspa`
+/// (note that the folder is hidden).
+///
+/// This must be called before using any other wallet
+/// SDK functions.
+///
+/// NOTE: This function will create a folder if it
+/// doesn't exist. This function will have no effect
+/// if invoked in the browser environment.
+/// @param {String} folder - the path to the storage folder
+#[wasm_bindgen(js_name = setDefaultStorageFolder, skip_jsdoc)]
+pub fn set_default_storage_folder(folder: String) -> Result<()> {
+    create_dir_all_sync(&folder).map_err(|err| Error::custom(format!("Failed to create storage folder: {err}")))?;
+    unsafe {
+        DEFAULT_STORAGE_FOLDER = Some(folder);
+    }
+    Ok(())
+}
```

Comment:
that's unsafe function, need to think about the invariant
also  we should use `leak()` function first, otherwise static str want be valid

Reply:
This was a misunderstanding as it is a self-contained `String`, so it won't leak.  As discussed separately, I've made a pass and added a few `SAFETY` comments: https://github.com/aspectron/rusty-kaspa/commit/f848da3b2901c08a6b12a6b30b35c2d4e69bf87c

Reply:
as discussed I've converted init functions to be `unsafe` and segregated them from JS bindings https://github.com/aspectron/rusty-kaspa/commit/d2628d8d417e72b47498e1b1b087e72c46174422